server/notifications.go

```diff
@@ -16,63 +16,65 @@ package server
 
 import (
 	"context"
-	"log"
 
 	"cloud.google.com/go/pubsub"
+	"github.com/apex/log"
 	"github.com/apigee/registry/rpc"
 	"google.golang.org/grpc/codes"
 	"google.golang.org/grpc/status"
 	"google.golang.org/protobuf/encoding/protojson"
 	"google.golang.org/protobuf/types/known/timestamppb"
 )
 
-const verbose = false
-
 const TopicName = "registry-events"
 
-var notificationTotal int
+func (s *RegistryServer) notify(ctx context.Context, change rpc.Notification_Change, resource string) {
+	logger := log.FromContext(ctx)
+	if !s.notifyEnabled {
+		return
+	}
 
-func (s *RegistryServer) notify(ctx context.Context, change rpc.Notification_Change, resource string) error {
-	if !s.notifyEnabled || s.projectID == "" {
-		return nil
+	if s.projectID == "" {
+		logger.Warn("Notifications are enabled but project ID is not set. Skipping notification.")
```

Comment:
Does this warn only once? If not, I think this could be very noisy.

Reply:
It will be noisy if the issue arises, but I think that's appropriate given the circumstance. Each warning log would correspond to one instance of unexpected behavior in the server, where the server is configured to send a notification but chooses not to try because it's missing the required project ID.

Currently though, our config validation should prevent the server from starting if PubSub is enabled without a project ID. This should only execute if we accidentally sneak past that validation in the future.

Reply:
Great, that initial config validation is what I would have suggested (but we already have it).